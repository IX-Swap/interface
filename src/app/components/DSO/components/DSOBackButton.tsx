import React from 'react'
import { AppRouterLinkComponent } from 'components/AppRouterLink'
import { Button } from '@material-ui/core'
import { IssuanceRoute } from 'app/pages/issuance/router/config'
import { useParams } from 'react-router-dom'

export const DSOBackButton = () => {
  const params = useParams()

  return (
    <Button
      component={AppRouterLinkComponent}
      to={IssuanceRoute.list} // TODO: add logic for the create DSO screen
      params={params}
    >
      Cancel
    </Button>
  )
}
