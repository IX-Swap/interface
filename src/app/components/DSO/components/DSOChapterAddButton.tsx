import React from 'react'
import { Button } from '@material-ui/core'

export interface DSOTeamAddButtonProps {
  append: Function
  text?: string
}

export const DSOChapterAddButton: React.FC<DSOTeamAddButtonProps> = props => {
  const { append, text = 'Add Team Member' } = props
  const handleClick = () => {
    append({})
  }

  return (
    <Button color='primary' variant='outlined' onClick={handleClick}>
      {text}
    </Button>
  )
}
