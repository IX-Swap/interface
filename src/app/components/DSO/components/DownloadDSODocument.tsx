import React from 'react'
import { useDownloadRawFile } from 'hooks/useDownloadRawFile'
import { IconButton, Tooltip } from '@mui/material'
import { convertBlobToFile, openFileInNewTab } from 'hooks/utils'
import { Launch } from '@mui/icons-material'

export interface DownloadDSODocumentProps {
  dsoId: string
  documentId: string
  type?: 'document' | 'subscriptionDocument'
}

export const DownloadDSODocument = (props: DownloadDSODocumentProps) => {
  const { dsoId, documentId, type = 'document' } = props
  const uri =
    type === 'document'
      ? `/issuance/dso/dataroom/documents/raw/${dsoId}/${documentId}`
      : `/issuance/dso/dataroom/subscription/raw/${dsoId}`
  const [download, { isLoading }] = useDownloadRawFile(uri, {
    onSuccess: ({ data }) => {
      const file = convertBlobToFile(data, '')
      openFileInNewTab(file)
    }
  })
  const handleDownload = async () => await download()

  return (
    <Tooltip title='Download File'>
      <IconButton disabled={isLoading} onClick={handleDownload} size='large'>
        <Launch color='disabled' style={{ width: 23, height: 23 }} />
      </IconButton>
    </Tooltip>
  )
}
