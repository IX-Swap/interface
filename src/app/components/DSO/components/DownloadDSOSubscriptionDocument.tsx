import React from 'react'
import { useDownloadRawFile } from 'hooks/useDownloadRawFile'
import { Button, ButtonProps } from '@mui/material'
import { convertBlobToFile, openFileInNewTab } from 'hooks/utils'

export interface DownloadDSOSubscriptionDocumentProps extends ButtonProps {
  dsoId: string
}

export const DownloadDSOSubscriptionDocument = (
  props: DownloadDSOSubscriptionDocumentProps
) => {
  const { dsoId, children = 'Download', ...rest } = props
  const uri = `/issuance/dso/dataroom/subscription/raw/${dsoId}`
  const [download, { isLoading }] = useDownloadRawFile(uri, {
    onSuccess: ({ data }) => {
      const file = convertBlobToFile(data, '')
      openFileInNewTab(file)
    }
  })
  const handleDownload = async () => await download()

  return (
    <Button {...rest} onClick={handleDownload} disabled={isLoading}>
      {children}
    </Button>
  )
}
