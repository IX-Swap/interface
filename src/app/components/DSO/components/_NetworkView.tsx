import React from 'react'
import { useAllNetworks } from 'app/pages/accounts/pages/withdrawalAddresses/hooks/useAllNetworks'

export interface NetworkViewProps {
  networkId: string
}

export const NetworkView = (props: NetworkViewProps) => {
  const { networkId } = props
  const { data, isLoading } = useAllNetworks()
  const network = data?.find(network => network._id === networkId)

  return isLoading ? <>'...' </> : <>{network?.name}</>
}
