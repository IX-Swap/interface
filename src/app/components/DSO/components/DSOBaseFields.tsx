import React from 'react'
import { Grid, TextField } from '@material-ui/core'
import { TypedField } from 'components/form/TypedField'
import { dateTimeValueExtractor, numericValueExtractor } from 'helpers/forms'
import { CorporateSelect } from 'components/form/CorporateSelect'
import { NetworkSelect } from 'components/form/NetworkSelect'
import { AssetSelect } from 'components/form/AssetSelect'
import { useFormContext } from 'react-hook-form'
import { DSOFormValues } from 'types/dso'
import { documentValueExtractor } from 'app/components/DSO/utils'
import { DataroomFileType } from 'config/dataroom'
import { DateTimePicker } from 'components/form/_DateTimePicker'
import { VSpacer } from 'components/VSpacer'
import { Dropzone } from 'components/dataroom/Dropzone'
import { CapitalStructureSelect } from 'components/form/CapitalStructureSelect'
import { NumericInput } from 'components/form/NumericInput'
import { positiveNumberFormat } from 'config/numberFormat'
import { FormSectionHeader } from 'app/components/DSO/components/FormSectionHeader'

export interface DSOBaseFieldsProps {
  isNew: boolean
  isLive: boolean
}

export const DSOBaseFields = (props: DSOBaseFieldsProps) => {
  const { isNew, isLive } = props
  const { control, watch } = useFormContext<DSOFormValues>()
  const capitalStructure = watch('capitalStructure')

  return (
    <Grid item>
      <VSpacer size='medium' />
      <Grid container direction='column' spacing={2}>
        <Grid item>
          <FormSectionHeader title='DSO Information' />
        </Grid>
        <Grid item>
          {/* @ts-ignore */}
          <TypedField
            customRenderer
            component={Dropzone}
            name='logo'
            label='Upload Logo'
            control={control}
            valueExtractor={documentValueExtractor}
            accept={DataroomFileType.image}
            documentInfo={{
              type: 'DSO Logo'
            }}
          />
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12} md={6}>
              <TypedField
                control={control}
                component={CapitalStructureSelect}
                label='Capital Structure'
                name='capitalStructure'
                helperText='Offering terms will be changed based on your capital structure'
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={6}>
              <TypedField
                component={NetworkSelect}
                label='Network'
                name='network'
                disabled={!isNew}
                control={control}
                helperText='Select your blockchain network from the list'
                variant='outlined'
              />
            </Grid>
          </Grid>
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12} md={6}>
              <TypedField
                component={TextField}
                label='Token Name'
                name='tokenName'
                disabled={isLive}
                control={control}
                helperText='Name of the token that describes your offering the best'
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={2}>
              <TypedField
                component={TextField}
                label='Symbol'
                name='tokenSymbol'
                disabled={isLive}
                control={control}
                helperText='Token symbol'
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={4}>
              <TypedField
                component={NumericInput}
                numberFormat={positiveNumberFormat}
                label='Decimal Places'
                name='decimalPlaces'
                disabled={capitalStructure?.toLowerCase() === 'equity'}
                control={control}
                helperText='Decimal Places'
                variant='outlined'
                valueExtractor={numericValueExtractor}
              />
            </Grid>
          </Grid>
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12} md={8}>
              <TypedField
                component={CorporateSelect}
                label='Corporate Structure'
                name='corporate'
                control={control}
                helperText='Select your corporate from the list'
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={4}>
              <TypedField
                assetType='Currency'
                component={AssetSelect}
                label='Currency'
                name='currency'
                control={control}
                helperText='Your preferred currency'
                variant='outlined'
              />
            </Grid>
          </Grid>
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12} md={6}>
              {/* @ts-ignore */}
              <TypedField
                component={DateTimePicker}
                customRenderer
                label='Launch Date'
                name='launchDate'
                control={control}
                disabled={isLive}
                valueExtractor={dateTimeValueExtractor}
                // @ts-expect-error
                defaultValue={null}
                helperText='Offering launch date'
                inputVariant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={6}>
              {/* @ts-ignore */}
              <TypedField
                component={DateTimePicker}
                customRenderer
                label='Completion Date'
                name='completionDate'
                control={control}
                valueExtractor={dateTimeValueExtractor}
                // @ts-expect-error
                defaultValue={null}
                helperText='Offering completion date'
                inputVariant='outlined'
              />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  )
}
