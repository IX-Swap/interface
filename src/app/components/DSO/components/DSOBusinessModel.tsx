import React from 'react'
import { TypedField } from 'components/form/TypedField'
import { RichTextEditor } from 'components/form/RichTextEditor'
import { wysiwygValueExtractor } from 'helpers/forms'
import { DSOContainer } from 'app/components/DSO/components/DSOContainer'
import { useFormContext } from 'react-hook-form'
import { DSOFormValues } from 'types/dso'

export const DSOBusinessModel = () => {
  const { control } = useFormContext<DSOFormValues>()

  return (
    <DSOContainer title='Business Model' item xs={12}>
      {/* @ts-ignore */}
      <TypedField
        control={control}
        component={RichTextEditor}
        customRenderer
        label='Business Model'
        name='businessModel'
        valueExtractor={wysiwygValueExtractor}
      />
    </DSOContainer>
  )
}
