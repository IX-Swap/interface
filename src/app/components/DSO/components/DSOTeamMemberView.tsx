import React from 'react'
import { Box, Grid } from '@mui/material'
import { DSOTeamMemberPhoto } from 'app/components/DSO/components/DSOTeamMemberPhoto'
import { renderStringToHTML } from 'app/components/DSO/utils'
import { DsoTeamMember } from 'types/dso'
import { Expandable } from 'app/components/Expandable/Expandable'
import { LabelledValue } from 'components/LabelledValue'

export interface DSOTeamMemberViewProps {
  dsoId: string
  member: DsoTeamMember
}

export const DSOTeamMemberView = (props: DSOTeamMemberViewProps) => {
  const { member, dsoId } = props

  return (
    <Expandable
      showArrow
      mainComponent={
        <Grid container direction='row' alignItems='center' spacing={2} p={1}>
          <Grid item>
            <DSOTeamMemberPhoto
              dsoId={dsoId}
              photoId={member.photo}
              size={48}
            />
          </Grid>
          <Grid item>
            <LabelledValue
              label={member.name}
              value={member.position}
              isRedesigned
              gap={0.2}
            />
          </Grid>
        </Grid>
      }
      expandedComponent={<Box px={2}>{renderStringToHTML(member.about)}</Box>}
    />
  )
}
