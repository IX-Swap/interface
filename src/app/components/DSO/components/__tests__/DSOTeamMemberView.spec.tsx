import { DSOTeamMemberView } from 'app/components/DSO/components/DSOTeamMemberView'
import React from 'react'

import { render, cleanup, fireEvent, waitFor } from 'test-utils'
import { dso } from '__fixtures__/authorizer'
import { teamMember } from '__fixtures__/issuance'

describe('DSOTeamMemberView', () => {
  afterEach(async () => {
    await cleanup()
    jest.clearAllMocks()
  })

  it('renders without errors', () => {
    render(<DSOTeamMemberView dsoId={dso._id} member={teamMember} />)
  })

  it('shows about when component is pressed', async () => {
    const { container, queryByText } = render(
      <DSOTeamMemberView dsoId={dso._id} member={teamMember} />
    )
    expect(queryByText('about')).toBeFalsy()
    fireEvent.click(container, { cancellable: true, bubbles: true })

    waitFor(() => {
      expect(queryByText('about')).toBeTruthy()
    })
  })

  it('matches snapshot', () => {
    const { container } = render(
      <DSOTeamMemberView dsoId={dso._id} member={teamMember} />
    )
    expect(container).toMatchSnapshot()
  })
})
