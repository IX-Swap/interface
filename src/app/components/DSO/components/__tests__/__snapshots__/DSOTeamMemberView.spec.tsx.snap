// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DSOTeamMemberView matches snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-1885q2t-MuiPaper-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-ofrcno-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-69x3yt-MuiAvatar-root"
              style="width: 48px; height: 48px;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-1ql7le6-MuiSvgIcon-root-MuiAvatar-fallback"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-vxrdkt-MuiTypography-root"
            >
              name
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1b4l2pg-MuiTypography-root"
            >
              position
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
